src/import/pull-job.c

```diff
@@ -236,7 +236,7 @@ static int pull_job_write_uncompressed(const void *p, size_t sz, void *userdata)
                 else
                         n = write(j->disk_fd, p, sz);
                 if (n < 0)
-                        return log_error_errno(errno, "Failed to write file: %m");
+                        return log_error_errno(j->allow_sparse ? (int) n : errno, "Failed to write file: %m");
```

Comment:
I'd prefer to convert the errno immediately above:
```C
     ...
else {
     n = write(j->disk_fd, p, sz);
     if (n < 0)
           n = -errno;
}
```
This has the advantage that it keeps the errno mess localized.